engine/rowset/src/main/java/io/deephaven/engine/rowset/impl/sortedranges/SortedRanges.java

```diff
@@ -590,14 +601,18 @@ public final boolean advanceImpl(final long v) {
                 nextRangeIdx = sar.count;
                 return true;
             }
-            final long after = sar.unpackedGet(p);
-            if (after < 0) {
-                currRangeEnd = -after;
+            cachedNextValue = sar.unpackedGet(p);
+            if (cachedNextValue < 0) {
+                currRangeEnd = -cachedNextValue;
                 nextRangeIdx = p + 1;
+                if (nextRangeIdx < sar.count) {
+                    cachedNextValue = sar.unpackedGet(nextRangeIdx);
+                }
                 return true;
             }
             currRangeEnd = currRangeStart;
             nextRangeIdx = p;
+            cachedNextValue = sar.unpackedGet(nextRangeIdx);
```

Comment:
I think this change is good overall.  What I am concerned about, but did not carefully run down is if there are cases like this where we will end up overflowing sar count or if we got all the potential off-by-one errors.  Presumably if the full test suite has passed, so I should get unnervous.